data/core/commands/findreplace.lua

```diff
@@ -193,6 +193,23 @@ local function select_next(reverse)
   if l2 then doc():set_selection(l2, c2, l1, c1) end
 end
 
+---@param in_selection? boolean
+local function find_replace(in_selection)
+  local l1, c1, l2, c2 = doc():get_selection()
+  local selected_text = ""
+  if not in_selection then
+    selected_text = doc():get_text(l1, c1, l2, c2)
```

Comment:
Didn't do that because of what you already mention, the entire selection getting added to a single line, which in the past for me was always deleted when replacing in selections. Also using the first line only doesn't seems much intuitive. Better behavior would be accomplished with the like of the search_ui which has more widgets/controls.

![20230130-154900](https://user-images.githubusercontent.com/1702572/215580465-1bb555c1-cedc-439b-a586-66be13066ded.png)